addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/HueEmulationServlet.java

```diff
@@ -0,0 +1,521 @@
+                "UTF-8");
+        String xml = String.format(discoString, InetAddress.getLocalHost().getHostAddress(),
+                System.getProperty("org.osgi.service.http.port"), getUDN());
+        resp.setContentType(APPLICATION_XML);
+        PrintWriter out = resp.getWriter();
+        out.write(xml);
+        out.close();
+    }
+
+    /**
+     * Returns a map of all our items that have voice tags.
+     *
+     * @param username
+     * @return
+     *         Map <item name, HueDevice>
+     */
+    private Map<String, HueDevice> getHueDevices() {
+        Collection<Item> items = getTagedItems();
+        Map<String, HueDevice> devices = new HashMap<String, HueDevice>();
+        Iterator<Item> it = items.iterator();
+        while (it.hasNext()) {
+            Item item = it.next();
+            devices.put(item.getName(), itemToDevice(item));
+        }
+        return devices;
+    }
+
+    /**
+     * Returns the item name and voice name of each item
+     *
+     * @param username
+     * @return
+     *         Map<item name, item voice tag>
+     */
+    public Map<String, String> getHueDeviceNames() {
+        Collection<Item> items = getTagedItems();
+        Map<String, String> devices = new HashMap<String, String>();
+        Iterator<Item> it = items.iterator();
+        while (it.hasNext()) {
+            Item item = it.next();
+            devices.put(item.getName(), item.getLabel());
+        }
+        return devices;
+    }
+
+    /**
+     * Converts an VoiceItem to a HueDevice
+     *
+     * @param voiceItem
+     * @return
+     *         HueDevice
+     */
+    private HueDevice itemToDevice(Item item) {
+        State itemState = item.getState();
+        short bri = 0;
+        if (itemState instanceof DecimalType) {
+            bri = (short) ((((DecimalType) itemState).intValue() * 255) / 100);
+        } else if (itemState instanceof OnOffType) {
+            bri = (short) (((OnOffType) itemState) == OnOffType.ON ? 255 : 0);
+        }
+        HueState hueState = new HueState(bri > 0, bri);
+        HueDevice d = new HueDevice(hueState, item.getLabel(), item.getName());
+        return d;
+    }
+
+    /**
+     * Gets all items that match our tag
+     *
+     * @return
+     */
+    private Collection<Item> getTagedItems() {
+        Collection<Item> items = new LinkedList<Item>();
+        for (Item item : itemRegistry.getItems()) {
+            for (String tag : item.getTags()) {
+                if (tag.indexOf(HOMEKIT_PREFIX) == 0) {
+                    items.add(item);
+                    break;
+                }
+            }
+        }
+        return items;
+    }
+
+    /**
+     * Checks if the username exists in our user file
+     *
+     * @param userName
+     * @return
+     * @throws IOException
+     */
+    private boolean authorizeUser(String userName) throws IOException {
+        synchronized (USER_FILE) {
```

Comment:
As this method is executed on every single call, you should really read the file only once at startup and keep the user list in a Set in memory for quick lookup.